### ISSUE TITLE
window.postMessage security vulnerability?

### ISSUE THREAD

**Issue by eirslett** [zalmoxisus/redux-devtools-extension#366]
Forgive me for posting, if this is a stupid question... I don't know a lot about the postMessage security details.
But I found some places in the code where it does `window.postMessage(message, '*');`. (like the function toContentScript).

Imagine this scenario;

1) A user is interacting with a trusted Redux-based web application, where sensitive information ends up in Redux (maybe the username/password as part of the login flow).
2) The user has Redux devtools extension installed and running (they are maybe a software developer working on the web application, or they're a user who is curious about what's going on under the hood in an interesting web application)
3) Redux devtools is enabled (also in production) for this trusted site.
4) The user is also visiting a malicious site in another browser tab.

Can the malicious site listen to `window.postMessage` events (intended for Redux devtools communication) that are broadcast on `*`, and gain access to the sensitive data? Or do the browsers have security policies that prevent this?

**Comment by zalmoxisus** [comment]
You're right, but I don't see other ways to pass data from client page script to extension's content script.

A malicious script can also redeclare `window. __REDUX_DEVTOOLS_EXTENSION__`, which is a global function. There are also [ways to get access to the `store` object](https://medium.com/@nuncamind/diving-deeper-into-twitters-redux-store-adventures-in-minified-vendor-javascript-67fbac5dc219) without using Redux DevTools.

So the recommendation here is not to store data like passwords in the states. For other sensitive data there're [`actionSanitizer` and `stateSanitizer`](https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#actionsanitizer--statesanitizer).

**Comment by kuhe** [comment]
The `*` could be replaced an array of configurable targets `string[]`.

The usage of `*` is causing an app malfunction in my case. However, I can overwrite the `window.postMessage` method as a workaround targeting only `localhost` and `@redux-devtools`.
